packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -96,106 +98,50 @@ export function cursorFromFieldData(
  * * `-0` =\> `+0`
  *
  * @param globalSchema - Schema for the whole tree for interperting `Any`.
- * @param typeSet - The set of types allowed by the parent context. Used to validate the input tree.
+ * @param allowedTypes - The set of types allowed by the parent context. Used to validate the input tree.
  */
-export function nodeDataToMapTree(
-	data: InsertableContent,
-	globalSchema: FlexTreeSchema,
-	typeSet: AllowedTypeSet,
-): MapTree {
+export function nodeDataToMapTree(data: InsertableContent, allowedTypes: AllowedTypes): MapTree {
 	assert(data !== undefined, 0x846 /* Cannot map undefined tree. */);
 
 	if (data === null) {
-		return valueToMapTree(data, globalSchema, typeSet);
+		return valueToMapTree(data, allowedTypes);
 	}
 	switch (typeof data) {
 		case "number":
 		case "string":
 		case "boolean":
-			return valueToMapTree(data, globalSchema, typeSet);
+			return valueToMapTree(data, allowedTypes);
 		default: {
 			if (isFluidHandle(data)) {
-				return valueToMapTree(data, globalSchema, typeSet);
+				return valueToMapTree(data, allowedTypes);
 			} else if (Array.isArray(data)) {
```

Comment:
The fact that nodeDataToMapTree branches based on the data and not based on the schema might be something we want to change.

For example, if I author a new schema, with a custom constructor that takes some different type that this code requires, the way our types work will expect data that this code will not handle.

Another way to look at it is the adding new fancy custom node types should work without having to update this code.
For example, when working on recursive schema, I made array nodes as well as maps take in an iterable instead of forcing an array or map. THat actuall alows data this code won't handle: I'd like future work like that to not break things.

That said, your change is progress we want, so no need to block it on such a refactoring, but eventually I do think nodeDataToMapTree should call into functionality provided on the schema (a method to check if the data is compatible, and a method to actually convert it to normalized tree data (ex: a map tree). I'm not exactual sure how that should work, but I think we will want something like that.

Reply:
Yeah, I'd been considering the same thing for the purpose of Map node construction. Theoretically we should be able to take in a Record for Map creation as well. I think I will leave that for a future PR, but I'll file something for it.

Reply:
Do we have a feature for tracking non-critical-path architectural improvements like this?

Reply:
I think they are tracked as tech-debt. I don't know how thats organized right now.

Reply:
Okay. For now I'll just file a task in the team-rohan board and plan to look at it next sprint. I don't expect it would require much effort.

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/7546